transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannel.java

```diff
@@ -369,19 +368,22 @@ private boolean doWriteMessage(Object msg) throws Exception {
         }
 
         final int writtenBytes;
-        if (data.hasMemoryAddress()) {
-            long memoryAddress = data.memoryAddress();
-            writtenBytes = socket.sendToAddress(memoryAddress, data.readerIndex(), data.writerIndex(),
-                    remoteAddress.getAddress(), remoteAddress.getPort());
-        } else if (data instanceof CompositeByteBuf) {
-            IovArray array = ((EpollEventLoop) eventLoop()).cleanArray();
-            array.add(data);
-            int cnt = array.count();
-            assert cnt != 0;
-
-            writtenBytes = socket.sendToAddresses(array.memoryAddress(0),
-                    cnt, remoteAddress.getAddress(), remoteAddress.getPort());
-        } else  {
+        if (PlatformDependent.hasUnsafe()) {
```

Comment:
you don't need to check this in the native transports ... they will not be available if unsafe is not available.